src/Framework/EventFile.cxx

```diff
@@ -231,6 +247,12 @@ void EventFile::setupEvent(Event *evt) {
   }
 }
 
+int EventFile::skipToEvent(int offset) {
+  ientry_ = offset % entries_;    // make sure the event number exists
```

Comment:
Should there be a `-1` here? So that `nextEvent()` moves us to the `offset` event?

Reply:
Yeah i thought about it but i thought it was a minor detail. You pick a number at random, and then you get that number + 1 in the end, it doesn't matter much. In the (unlikely but occurring) event that you picked the last event number, the tree will wrap. So nothing changes in practice. The shift is arbitrary but deterministic.

Reply:
Implemented this since I was anyway updating the file.